code/modules/antagonists/traitor/equipment/Malf_Modules.dm

```diff
@@ -701,6 +701,8 @@ GLOBAL_LIST_INIT(malf_modules, subtypesof(/datum/ai_module))
 		if(!is_station_level(F.z))
 			continue
 		F.obj_flags |= EMAGGED
+		for(var/obj/machinery/door/firedoor/firelock in F.my_area.firedoors)
+			firelock.obj_flags |= EMAGGED
```

Comment:
Should this still be done for disabling thermal sensors?

Reply:
Yes. I've redone how emagging works; emagging a fire alarm now just disables the noise. A doorjack is needed to emag firelocks, and that will disable the firelock's detection and put it into a state where it will never close except when forced (via silicon command or crowbar).

That said, I probably need to make sure the malf module runs all the emag code now, yeah.

Reply:
It now calls the firelock's emag_act proc instead. Also, it finds firelocks by looking through the GLOB.machines list, rather than checking the list of firelocks in each fire alarm from the prior loop (which I suspect will be faster solely due to not dealing with duplicate entries).